src/middlewared/middlewared/plugins/disk_/configure_swap.py

```diff
@@ -0,0 +1,223 @@
+import os
+import platform
+import subprocess
+
+from collections import defaultdict
+
+from middlewared.service import CallError, private, Service
+from middlewared.utils import Popen, run
+
+IS_LINUX = platform.system().lower() == 'linux'
+MIRROR_MAX = 5
+
+
+class DiskService(Service):
+
+    @private
+    async def swaps_configure(self):
+        """
+        Configures swap partitions in the system.
+        We try to mirror all available swap partitions to avoid a system
+        crash in case one of them dies.
+        """
+        used_partitions_in_mirror = set()
+        create_swap_devices = {}
+        disks = [i async for i in await self.middleware.call('pool.get_disks')]
+        disks.extend(await self.middleware.call('boot.get_disks'))
+        existing_swap_devices = {'mirrors': [], 'partitions': []}
+        mirrors = await self.middleware.call('disk.get_mirrors')
+        encrypted_mirrors = {m['encrypted_provider']: m for m in mirrors if m['encrypted_provider']}
+        all_partitions = {p['name']: p for p in await self.middleware.call('disk.list_all_partitions')}
+
+        for device in await self.middleware.call('disk.get_swap_devices'):
+            if device in encrypted_mirrors or device.startswith(('/dev/md', 'mirror/')):
+                # This is going to be complete path for linux and mirror/swapname.eli for freebsd
+                existing_swap_devices['mirrors'].append(device)
+            else:
+                existing_swap_devices['partitions'].append(device)
+
+        # disk.get_mirrors is going to get us complete path for linux and mirror/swapname for freebsd
+        # point to note is that we don't get .eli suffix from above - so we have to be careful here and
+        # use encrypted_provider instead
+        for mirror in mirrors:
+            # If the mirror is degraded or disk is not in a pool lets remove it
+            mirror_name = (mirror['encrypted_provider'] or mirror['real_path']) if IS_LINUX else (
+                mirror['encrypted_provider'] or mirror['path']
+            ).strip('/dev/')
```

Comment:
`strip` does not work this way, also, can you please use regular if / else block?